File: DictionaryProject/WordBuilder/TokenSet.vb

Public Class TokenSet

    Private _Name As String
    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property

    Private _Tokens As New List(Of String)()
    Public ReadOnly Property Tokens() As List(Of String)
        Get
            Return _Tokens
        End Get
    End Property

    Private Shared _Random As New Random()

    Public Function GetToken() As String
        Return Tokens(_Random.Next(0, Tokens.Count))
    End Function
End Class

File: monowordbuilder/WordBuilderProject/TokenSet.cs
using System;
using System.Collections.Generic;

public class TokenSet
{
    
	private string _Name;
	public string Name {
		get { return _Name; }
		set { _Name = value; }
	}
    
	private List<String> _Tokens = new List<string>();
	public List<String> Tokens {
		get { return _Tokens; }
	}
    
	private static Random _Random = new Random();
    
	public string GetToken()
	{
		return Tokens[_Random.Next(0, Tokens.Count)];
	}
}